dataRetriever/dataPool/proofsCache/proofsBucket.go

```diff
@@ -0,0 +1,35 @@
+package proofscache
+
+import "github.com/multiversx/mx-chain-core-go/data"
+
+type proofNonceBucket struct {
+	maxNonce      uint64
+	proofsByNonce []*proofNonceMapping
+	bucketSize    int
+}
+
+func newProofBucket(bucketSize int) *proofNonceBucket {
+	return &proofNonceBucket{
+		proofsByNonce: make([]*proofNonceMapping, 0),
+		bucketSize:    bucketSize,
+	}
+}
+
+func (p *proofNonceBucket) size() int {
+	return len(p.proofsByNonce)
+}
+
+func (p *proofNonceBucket) isFull() bool {
```

Comment:
I would use fixed size and deterministic range for each bucket.
e.g a proof with nonce x should be added into the bucket with first nonce x/bucketSize * bucketSize (integer division)

Reply:
changed to use fixed size buckets